pkg/resources/table.go

```diff
@@ -485,126 +454,198 @@ func getPrimaryKey(from interface{}) (to primarykey) {
+		// TODO [SNOW-867240]: SHOW TABLE returns last part of id without double quotes... we have to quote it again. Move it to SDK.
+		sequenceIdRaw := strings.TrimSuffix(defaultRaw, ".NEXTVAL")
+		def["sequence"] = sdk.NewSchemaObjectIdentifierFromFullyQualifiedName(sequenceIdRaw).FullyQualifiedName()
+		return def
+	}
+
+	if strings.Contains(defaultRaw, "(") && strings.Contains(defaultRaw, ")") {
+		def["expression"] = defaultRaw
+		return def
+	}
+
+	columnType := strings.ToUpper(string(td.Type))
+	if strings.Contains(columnType, "CHAR") || columnType == "STRING" || columnType == "TEXT" {
+		def["constant"] = snowflake.UnescapeSnowflakeString(defaultRaw)
+		return def
+	}
+
+	if toColumnIdentityConfig(td) != nil {
+		/*
+			Identity/autoincrement information is stored in the same column as default information. We want to handle the identity separate so will return nil
+			here if identity information is present. Default/identity are mutually exclusive
+		*/
+		return nil
+	}
+
+	def["constant"] = defaultRaw
+	return def
+}
+
+func toColumnIdentityConfig(td sdk.TableColumnDetails) map[string]any {
+	// if autoincrement is used this is reflected back IDENTITY START 1 INCREMENT 1
+	if td.Default == nil {
+		return nil
+	}
+
+	defaultRaw := *td.Default
+
+	if strings.Contains(defaultRaw, "IDENTITY") {
+		identity := map[string]any{}
+
+		split := strings.Split(defaultRaw, " ")
+		start, _ := strconv.Atoi(split[2])
+		step, _ := strconv.Atoi(split[4])
+
+		identity["start_num"] = start
+		identity["step_num"] = step
+
+		return identity
+	}
+	return nil
 }
 
 // CreateTable implements schema.CreateFunc.
 func CreateTable(d *schema.ResourceData, meta interface{}) error {
 	db := meta.(*sql.DB)
-	database := d.Get("database").(string)
-	schema := d.Get("schema").(string)
+	ctx := context.Background()
+	client := sdk.NewClientFromDB(db)
+
+	databaseName := d.Get("database").(string)
+	schemaName := d.Get("schema").(string)
 	name := d.Get("name").(string)
+	id := sdk.NewSchemaObjectIdentifier(databaseName, schemaName, name)
 
-	columns := getColumns(d.Get("column").([]interface{}))
+	tableColumnRequests := getTableColumnRequests(d.Get("column").([]interface{}))
 
-	builder := snowflake.NewTableWithColumnDefinitionsBuilder(name, database, schema, columns.toSnowflakeColumns())
+	createRequest := sdk.NewCreateTableRequest(id, tableColumnRequests)
 
-	// Set optionals
 	if v, ok := d.GetOk("comment"); ok {
-		builder.WithComment(v.(string))
+		createRequest.WithComment(sdk.String(v.(string)))
 	}
 
 	if v, ok := d.GetOk("cluster_by"); ok {
-		builder.WithClustering(expandStringList(v.([]interface{})))
+		createRequest.WithClusterBy(expandStringList(v.([]interface{})))
 	}
 
 	if v, ok := d.GetOk("primary_key"); ok {
-		pk := getPrimaryKey(v.([]interface{}))
-		builder.WithPrimaryKey(pk.toSnowflakePrimaryKey())
+		keysList := v.([]any)
+		if len(keysList) > 0 {
+			keyName := keysList[0].(map[string]any)["name"].(string)
```

Comment:
Step 7 was validating the empty name for the key (so it was working correctly with such impl). I changed test 7 config to not even specify this empty name - it was still working. I changed the impl either way - why not.

Reply:
Good, If the test set's primary key and in step 7 it's removed or reassigned the memory is already initialized (I didn't analyze the test). I was wondering what would happen in the initial state when creating a table, because of the initialized/uninitialized memory cases I've had before.